File: app/src/main/java/com/bmco/cratesiounofficial/TrendingPageAdapter.java
package com.bmco.cratesiounofficial;

import android.content.Context;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.bmco.cratesiounofficial.fragments.JustUpdatedPageFragment;
import com.bmco.cratesiounofficial.fragments.MostDownloadedPageFragment;
import com.bmco.cratesiounofficial.fragments.NewCratesPageFragment;
import com.bmco.cratesiounofficial.fragments.PopularCategoriesPageFragment;
import com.bmco.cratesiounofficial.fragments.PopularKeywordsPageFragment;

/**
 * Created by Bertus on 25-5-2017.
 */

public class TrendingPageAdapter extends FragmentPagerAdapter {
    final int PAGE_COUNT = 3;
    private String tabTitles[] = new String[] { "New Crates", "Most Downloaded", "Just Updated", "Popular Keywords", "Popular Categories" };
    private Context context;

    public TrendingPageAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public android.support.v4.app.Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new NewCratesPageFragment();
            case 1:
                return new MostDownloadedPageFragment();
            case 2:
                return new JustUpdatedPageFragment();
            case 3:
                return PopularKeywordsPageFragment.newInstance(position);
            case 4:
                return PopularCategoriesPageFragment.newInstance(position);
            default:
                return null;
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        // Generate title based on item position
        return tabTitles[position];
    }
}

File: app/src/main/java/com/bmco/cratesiounofficial/TrendingPageAdapter.kt
package com.bmco.cratesiounofficial


import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.bmco.cratesiounofficial.fragments.JustUpdatedPageFragment
import com.bmco.cratesiounofficial.fragments.MostDownloadedPageFragment
import com.bmco.cratesiounofficial.fragments.NewCratesPageFragment
import com.bmco.cratesiounofficial.fragments.PopularCategoriesPageFragment
import com.bmco.cratesiounofficial.fragments.PopularKeywordsPageFragment

/**
 * Created by Bertus on 25-5-2017.
 */

class TrendingPageAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
    private val pageCount = 3
    private val tabTitles = arrayOf("New Crates", "Most Downloaded", "Just Updated", "Popular Keywords", "Popular Categories")

    override fun getCount(): Int {
        return pageCount
    }

    override fun getItem(position: Int): Fragment? {
        return when (position) {
            0 -> NewCratesPageFragment()
            1 -> MostDownloadedPageFragment()
            2 -> JustUpdatedPageFragment()
            3 -> PopularKeywordsPageFragment.newInstance(position)
            4 -> PopularCategoriesPageFragment.newInstance(position)
            else -> null
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        // Generate title based on item position
        return tabTitles[position]
    }
}
